File: Leetcode100/leetcode100_python/ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        p, q, r = 1, 1, 1
        for i in range(2, n + 1):
            r = p + q
            p = q
            q = r
        return r

if __name__ == "__main__":
    n = int(input())
    solution = Solution()
    print(solution.climbStairs(n))

File: Leetcode100/leetcode100/src/main/java/com/jiang/leetcode100/climbing_stairs/ClimbingStairs.java
package com.jiang.leetcode100.climbing_stairs;

import java.util.Scanner;

class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n + 1];
        dp[1] = 1;
        dp[0] = 1;
        for(int i = 2; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n];
    }
}

public class ClimbingStairs {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        scanner.close();
        System.out.println(new Solution().climbStairs(n));
    }
}